planning/behavior_path_planner/src/planner_manager.cpp

```diff
@@ -347,20 +364,52 @@ std::pair<SceneModulePtr, BehaviorModuleOutput> PlannerManager::runRequestModule
    * remove non-executable modules.
    */
   for (const auto & module_ptr : sorted_request_modules) {
-    if (!getManager(module_ptr)->isSimultaneousExecutableAsCandidateModule()) {
-      if (executable_modules.empty()) {
-        executable_modules.push_back(module_ptr);
-        break;
-      }
+    // if this module is always executable, add it to the list immediately.
+    if (getManager(module_ptr)->isAlwaysExecutableModule()) {
+      executable_modules.push_back(module_ptr);
+      continue;
     }
 
-    const auto itr =
-      std::find_if(executable_modules.begin(), executable_modules.end(), [this](const auto & m) {
-        return !getManager(m)->isSimultaneousExecutableAsCandidateModule();
-      });
+    // Condition 3: If the executable modules are either empty or consist only of always-executable
+    // modules, any module can be added.
+    const bool has_non_always_executable_module = std::any_of(
+      executable_modules.begin(), executable_modules.end(),
+      [this](const auto & m) { return !getManager(m)->isAlwaysExecutableModule(); });
+    if (!has_non_always_executable_module) {
+      executable_modules.push_back(module_ptr);
+      continue;
+    }
+
+    // Condition 2: Only modules that are always executable can be added
+    // if there exists at least one executable module that is neither always nor simultaneous
+    // executable.
+    const auto find_block_module = [this](const auto & m) {
+      return !getManager(m)->isAlwaysExecutableModule() &&
+             !getManager(m)->isSimultaneousExecutableAsApprovedModule();
+    };
+    const auto itr_block =
+      std::find_if(executable_modules.begin(), executable_modules.end(), find_block_module);
+    if (
+      itr_block != executable_modules.end() && getManager(module_ptr)->isAlwaysExecutableModule()) {
+      executable_modules.push_back(module_ptr);
+      continue;
+    }
 
-    if (itr == executable_modules.end()) {
+    // Condition 1: Only modules that are either always executable or simultaneous executable can be
+    // added if there exists at least one executable module that is simultaneous but not always
+    // executable.
+    const auto find_non_always_simultaneous_module = [this](const auto & m) {
+      return !getManager(m)->isAlwaysExecutableModule() &&
+             getManager(m)->isSimultaneousExecutableAsApprovedModule();
+    };
+    const auto itr_non_always_simultaneous = std::find_if(
+      executable_modules.begin(), executable_modules.end(), find_non_always_simultaneous_module);
+    if (
+      itr_non_always_simultaneous != executable_modules.end() &&
+      (getManager(module_ptr)->isAlwaysExecutableModule() ||
+       getManager(module_ptr)->isSimultaneousExecutableAsApprovedModule())) {
       executable_modules.push_back(module_ptr);
+      continue;
```

Comment:
necessary?

Reply:
no, fixed in [0071bbd](https://github.com/autowarefoundation/autoware.universe/pull/4785/commits/0071bbdd8aa4d510a67f6d9bc32e23716f7aa416)